### PR TITLE
Update Task Thin Executions list function for parity with 2.11.x

### PR DESCRIPTION
**Pull request by corneil**
Updated task thin executions link handing and added extra test for parity with 2.11.x

See #6062

### CODE REVIEW

**Review by cppwfs — CHANGES_REQUESTED** [review]
Looks good just a couple of nitpicks

Since you are bringing this class up to date can you remove the following since they are not being used:
* the AboutResource import 
* actualDataFlowServerCoreVersion
* aboutLink

Thanks!

File: spring-cloud-dataflow-rest-client/src/test/java/org/springframework/cloud/dataflow/rest/client/TaskTemplateTests.java [hunk 1]
```diff
@@ -47,21 +47,14 @@ void setup() {
...
 
-	@Test
-	void oldDataFlow() {
-		validateExecutionLinkNotPresent("1.6.0");
-	}
-
 	@Test
 	void minDataFlow() {
-		validateExecutionLinkPresent("1.7.0");
+		validateExecutionLinkPresent("3.0.0");
```
[the remark this answers is not in the document]
Comment (corneil) [reply spring-cloud-dataflow-rest-client/src/test/java/org/springframework/cloud/dataflow/rest/client/TaskTemplateTests.java:52]: Until the next release we will be needing those again for the versions we support.

File: spring-cloud-dataflow-rest-client/src/test/java/org/springframework/cloud/dataflow/rest/client/TaskTemplateTests.java [hunk 2]
```diff
@@ -47,21 +47,14 @@ void setup() {
...
 	}
 
 	@Test
 	void futureDataFlow() {
-		validateExecutionLinkPresent("1.8.0");
-		validateExecutionLinkPresent("1.9.0");
-		validateExecutionLinkPresent("2.0.0");
+		validateExecutionLinkPresent("3.0.0");
```
[the remark this answers is not in the document]
Comment (corneil) [reply spring-cloud-dataflow-rest-client/src/test/java/org/springframework/cloud/dataflow/rest/client/TaskTemplateTests.java:57]: At the moment we don't have old versions we support with links missing

File: spring-cloud-dataflow-rest-client/src/test/java/org/springframework/cloud/dataflow/rest/client/TaskTemplateTests.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (cppwfs) [reply spring-cloud-dataflow-rest-client/src/test/java/org/springframework/cloud/dataflow/rest/client/TaskTemplateTests.java:52]: Can you add a TODO to let the future coders know that this needs to be updated.

File: spring-cloud-dataflow-rest-client/src/test/java/org/springframework/cloud/dataflow/rest/client/TaskTemplateTests.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (corneil) [reply spring-cloud-dataflow-rest-client/src/test/java/org/springframework/cloud/dataflow/rest/client/TaskTemplateTests.java:62]: Added

**Comment by cppwfs** [comment]
Squashed, Rebased, Merged
@corneil Thanks for the contribution!!!!

**Pull request closed without merging**